Update README.md (#15)
slidoapp/dbt-coverage: README.md

# dbt-coverage

<a href="https://pypi.org/project/dbt-coverage/"><img alt="PyPI" src="https://img.shields.io/pypi/v/dbt-coverage"></a>
<a href="https://pepy.tech/project/dbt-coverage"><img alt="Downloads" src="https://pepy.tech/badge/dbt-coverage"></a>
![GitHub last commit](https://img.shields.io/github/last-commit/slidoapp/dbt-coverage)
![PyPI - Python Version](https://img.shields.io/pypi/pyversions/dbt-coverage)
![PyPI - Format](https://img.shields.io/pypi/format/dbt-coverage)
<a href="https://github.com/slidoapp/dbt-coverage/blob/main/LICENSE.md"><img alt="License: MIT" src="https://img.shields.io/github/license/slidoapp/dbt-coverage"></a>
[![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fslidoapp%2Fdbt-coverage.svg?type=shield)](https://app.fossa.com/projects/git%2Bgithub.com%2Fslidoapp%2Fdbt-coverage?ref=badge_shield)

> _One-stop-shop for docs and test coverage of [`dbt`](https://github.com/dbt-labs/dbt) projects._

Author: @sweco I guess it would make sense to use this tagline a bit more 🙂

Author: No no, that was about it 🙂

## Why do I need something like this?

_**[`dbt-coverage`](https://github.com/slidoapp/dbt-coverage) is to [`dbt`](https://github.com/dbt-labs/dbt) what [`coverage.py`](https://github.com/nedbat/coveragepy) and [`interrogate`](https://interrogate.readthedocs.io/en/latest/) are to Python.**_

> It is a single CLI tool which checks your `dbt` project for missing docstrings and tests.

Reviewer: Maybe not docstrings but documentation or docs? I believe docstring is a Python-specific documentation string.

Author: @sweco you are right, let me update it

> Keeping tests and documentation close to the actual SQL code that generates the final model is one of the best design choices of `dbt`. It ensures tests are actually used and documentation is actually useful. But how do you make adding those a habit in your [`dbt`](https://github.com/dbt-labs/dbt) project?

Reviewer: I'm just thinking that maybe we can keep the order always the same? First docs and then tests?

```suggestion
Keeping documentation and tests close to the actual SQL code that generates the final model is one of the best design choices of `dbt`. It ensures documentation is actually useful and tests are actually used. But how do you make adding those a habit in your [`dbt`](https://github.com/dbt-labs/dbt) project?
```

Author: Sure, let's do that!

That is exactly where `dbt-coverage` comes in. It will

> - Give you a better sense of how well is your project documented and covered with tests;

Reviewer: I'm not sure about this but shouldn't the word order maybe be inversed?

```suggestion
- Give you a better sense of how well your project is documented and covered with tests;
```

Author: @sweco I'll probably rewrite it completely

- Help your CI/CD pipeline make sure new changes include documentation and tests;

> - Let you quickly asses the documentatio and tests of a new `dbt` project you get your hands on.

Reviewer: ```suggestion
- Let you quickly assess the documentation and tests of a new `dbt` project you get your hands on.
```

> - Let you quickly asses the documentatio and tests of a new `dbt` project you get your hands on.

Reviewer: @mrshu, I'm not sure I got this correctly. Does it mean that thanks to `dbt-coverage` we can quickly know how well a new project is documented and tested?

Author: @sweco yes, that's exactly what I've had in mind 🙂

## Related packages

- https://github.com/mikaelene/dbt-test-coverage
- [interrogate](https://interrogate.readthedocs.io/en/latest/) (docs coverage for Python)
- [coverage.py](https://github.com/nedbat/coveragepy) (execution coverage for Python)

## License

Licensed under the MIT license (see [LICENSE.md](LICENSE.md) file for more
details).

> [![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fslidoapp%2Fdbt-coverage.svg?type=large)](https://app.fossa.com/projects/git%2Bgithub.com%2Fslidoapp%2Fdbt-coverage?ref=badge_large)

Reviewer: This badge is really nice, I'm just thinking whether we shouldn't add some heading above. Maybe Fossa?

```suggestion
## Fossa

[![FOSSA Status](https://app.fossa.com/api/projects/git%2Bgithub.com%2Fslidoapp%2Fdbt-coverage.svg?type=large)](https://app.fossa.com/projects/git%2Bgithub.com%2Fslidoapp%2Fdbt-coverage?ref=badge_large)
```

Author: @sweco I don't think it needs that -- the link should be enough 🙂

Reviewer: I just think that now it seems like a part of the License. Or you don't think it's an issue?

![image](https://user-images.githubusercontent.com/11132999/145080874-a78ac360-fad1-4890-adf1-aaff37b9b1bb.png)

Author: @sweco well, some use it as a full replacement of the License part, so in that sense, I don't find it to be an issue.

https://github.com/codecov/python-standard/blob/master/README.md#license

That said, I don't mind adding a new header, I just didn't find it to be necessary 🙂

Reviewer: > some use it as a full replacement of the License part

Ah, I see! 😊  Thanks for the explanation, no need to add it.
